testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java

```diff
@@ -474,6 +474,23 @@ public void testLocationHeaderInResponse() {
                         });
     }
 
+    @Test
+    public void equalsSignInRedirectUrl() {
+
+        testAppUrl = authServerContextRootPage.toString().replace("localhost", NIP_IO_URL) + JAVASCRIPT_URL + "/index.html?test=bla=bla&super=man";
```

Comment:
It's done [here](https://github.com/keycloak/keycloak/blob/303514c73fd5b8fafcb9602c73f4b6fdafc6774a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java#L508) as well. How do I set the `testAppUrl` by using `AuthServerTestEnricher.getAuthServerContextRoot` method?

Reply:
I see. It is working because testAppUrl is assigned in `@Before` method, so it is same for each test. However it is useless to reassign it before each method since for majority of methods it is same. We can have something like this
```
private static final String TEST_APP_URL = AuthServerTestEnricher.getAuthServerContextRoot().replace("localhost", NIP_IO_URL) + JAVASCRIPT_URL + "/index.html";
```
instead of
```
private String testAppUrl;
```
and then create a new variable based on TEST_APP_URL everywhere we need to change it. 

However this is little bit out of this issue and the test will be working even without it, so I won't insist on changing it :-). Feel free to ignore this comment.

Reply:
Right, I feel like this is a separate issue